src/librustc_errors/lib.rs

```diff
@@ -786,7 +786,7 @@ impl HandlerInner {
                 .emitted_diagnostic_codes
                 .iter()
                 .filter_map(|x| match &x {
-                    DiagnosticId::Error(s) if registry.find_description(s).is_some() => {
+                    DiagnosticId::Error(s) if registry.find_description(s).has_value() => {
```

Comment:
Is there a better way to do this here? I tried doing a match but couldn't quite get it to work. Is this okay?